src/providers/app-config/app-config.ts

```diff
@@ -135,33 +140,43 @@ export class AppConfigProvider {
   private getRemoteData = (): Promise<any> =>
     new Promise((resolve, reject) => {
       if (this.networkState.getNetworkState() === ConnectionStatus.ONLINE) {
-        this.httpClient.get<any>(this.environmentFile.configUrl)
-          .pipe(timeout(30000))
-          .subscribe(
-            (data) => {
-              this.dataStore.setItem('CONFIG', JSON.stringify(data));
-              resolve(data);
-            },
-            (error) => {
-              this.store$.dispatch(new SaveLog(this.logHelper
-                .createLog(LogType.ERROR, 'Getting remote config failed, using cached data', error)));
-              reject(error);
-              if (error !== AuthenticationError.USER_NOT_AUTHORISED) {
-                this.getCachedRemoteConfig()
-                  .then(data => resolve(data))
-                  .catch(error => reject(error));
-              } else {
-                reject(error);
-              }
-            },
+        this.appInfoProvider.getMajorAndMinorVersionNumber()
+        .then((version:string) => {
+          const url = `${this.environmentFile.configUrl}?app_version=${version}`;
+          this.httpClient.get<any>(url)
+            .pipe(timeout(30000))
+            .subscribe(
+              (data) => {
+                this.dataStore.setItem('CONFIG', JSON.stringify(data));
+                resolve(data);
+              },
+              (error: HttpErrorResponse) => {
+                if (this.shouldGetCachedConfig(error.error)) {
+                  this.logError('Getting remote config failed, using cached data', error.error);
+                  this.getCachedRemoteConfig()
+                    .then(data => resolve(data))
+                    .catch(error => reject(error));
+                } else {
+                  this.logError('Getting remote config failed, not using cached data', error.error);
+                  reject(error);
+                }
+              },
           );
+        });
       } else {
         this.getCachedRemoteConfig()
           .then(data => resolve(data))
           .catch(error => reject(error));
       }
     })
 
+  private shouldGetCachedConfig = (errorMessage: string): boolean =>
+    errorMessage  !== AuthenticationError.USER_NOT_AUTHORISED &&
+    errorMessage !== AppConfigError.INVALID_APP_VERSION
```

Comment:
I know it's valid syntax - but i thought coding standards are to have ; on the end... (and i thought there were linting rules around this)

Reply:
liniting told me to take of the ;